src/librustdoc/html/highlight.rs

```diff
@@ -1223,17 +1242,18 @@ impl<'src> Classifier<'src> {
                 LiteralKind::Float { .. } | LiteralKind::Int { .. } => Class::Number,
             },
             TokenKind::GuardedStrPrefix => return no_highlight(sink),
-            TokenKind::Ident | TokenKind::RawIdent
-                if self.peek_non_whitespace() == Some(TokenKind::Bang) =>
-            {
-                self.in_macro = true;
-                let span = new_span(before, text, file_span);
-                sink(DUMMY_SP, Highlight::EnterSpan { class: Class::Macro(span) });
-                sink(span, Highlight::Token { text, class: None });
+            TokenKind::RawIdent if self.peek_non_whitespace() == Some(TokenKind::Bang) => {
```

Comment:
might be missing something and maybe too lazy to dig in any deeper, but why did you remove `TokenKind::Ident` from this arm?

Reply:
Because a `RawIdent` can never be a keyword, meaning `get_real_ident_class` is unneeded, allowing this check to be simpler.